src/dynrpg_textplugin.cpp

```diff
@@ -0,0 +1,494 @@
+	}
+
+	void AddText(const std::string& text) {
+		if (texts.empty()) {
+			texts.push_back(text);
+		} else {
+			texts.back() += text;
+		}
+
+		Refresh();
+	}
+
+	void ClearText() {
+		texts.clear();
+
+		Refresh();
+	}
+
+	void SetPosition(int new_x, int new_y) {
+		x = new_x;
+		y = new_y;
+	}
+
+	void SetColor(int new_color) {
+		color = new_color;
+
+		Refresh();
+	}
+
+	void SetPictureId(int new_pic_id) {
+		pic_id = new_pic_id;
+
+		Refresh();
+	}
+
+	void SetFixed(bool fixed) {
+		this->fixed = fixed;
+	}
+
+	void Draw(Bitmap& dst) override {
+		if (!bitmap) {
+			return;
+		}
+
+		const Game_Pictures::Picture* pic = Main_Data::game_pictures->GetPicturePtr(pic_id);
+		if (!pic) {
+			return;
+		}
+		const Sprite_Picture* sprite = pic->sprite;
+		if (!sprite) {
+			return;
+		}
+
+		// For unknown reasons the official plugin has an y-offset of 2
+		if (fixed) {
+			dst.Blit(x - Game_Map::GetDisplayX() / 16, y - Game_Map::GetDisplayY() / 16 + 2, *bitmap, bitmap->GetRect(), sprite->GetOpacity());
+		} else {
+			dst.Blit(x, y + 2, *bitmap, bitmap->GetRect(), sprite->GetOpacity());
+		}
+	};
+
+	void Update() {
+		const Game_Pictures::Picture* pic = Main_Data::game_pictures->GetPicturePtr(pic_id);
+		if (!pic) {
+			return;
+		}
+		const Sprite_Picture* sprite = pic->sprite;
+		if (!sprite) {
+			return;
+		}
+
+		if (z != sprite->GetZ()) {
+			z = sprite->GetZ() + 1;
+			SetZ(z);
+		}
+	}
+
+	std::vector<uint8_t> Save(const std::string& id) {
+		std::stringstream ss;
+		ss << x << "," << y << ",";
+		for (int i = 0; i < texts.size(); ++i) {
+			std::string t = texts[i];
+			// Replace , with a sentinel 0x01 to not mess up the tokenizer
+			std::replace(t.begin(), t.end(), ',', '\1');
+			ss << t;
+			if (i < texts.size() - 1) {
+				ss << "\n";
+			}
+
+		}
+		ss << "," << color << "," << id;
+
+		ss << "," << 255 << "," << (fixed ? "1" : "0") << "," << pic_id;
+
+		std::vector<uint8_t> data;
+
+		std::string s = ss.str();
+		size_t slen = s.size();
+
+		data.resize(slen);
+		data.insert(data.end(), s.begin(), s.end());
+
+		return data;
+	}
+
+	static DynRpgText* GetTextHandle(const std::string& id, bool silent = false) {
+		PendingMessage pm;
+		pm.PushLine(id, DynRpgText::CommandCodeInserter);
+		std::string new_id = pm.GetLines().front();
+
+		auto it = graphics.find(new_id);
+		if (it == graphics.end()) {
+			if (!silent) {
+				Output::Warning("No text with ID %s found", new_id.c_str());
+			}
+			return nullptr;
+		}
+
+		return (*it).second.get();
+	}
+
+	static std::string CommandCodeInserter(char ch, const char** iter, const char* end, uint32_t escape_char) {
+		if (ch == 'N' || ch == 'n') {
+			return PendingMessage::DefaultCommandInserter('n', iter, end, escape_char);
+		} else if (ch == 'V' || ch == 'v') {
+			return PendingMessage::DefaultCommandInserter('v', iter, end, escape_char);
+		} else if (ch == 'I' || ch == 'i') {
+			auto parse_ret = Game_Message::ParseParamImpl('I', 'i', *iter, end, escape_char, true);
```

Comment:
@fmatthew5876 
Would be nice to have ``ParseParamImpl`` exported. This provides a very simple way to extract values. But maybe use a different name (no Impl?).